app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt

```diff
@@ -66,42 +106,27 @@ class AuthenticationActivity : AppCompatActivity(), HasSupportFragmentInjector {
         return super.onOptionsItemSelected(item)
     }
 
-    private fun loadCredentials() {
-        intent.getLoginDeepLinkInfo()?.let {
-            showServerFragment(it)
-        }.ifNull {
-            val newServer = intent.getBooleanExtra(INTENT_ADD_NEW_SERVER, false)
-            presenter.loadCredentials(newServer) { isAuthenticated ->
-                if (isAuthenticated) {
-                    showChatList()
-                } else {
-                    showOnBoardingFragment()
-                }
+    private fun routeDeepLink(deepLinkInfo: DeepLinkInfo) {
+        presenter.loadCredentials(false) { isAuthenticated ->
+            if (isAuthenticated) {
+                presenter.toChatList(deepLinkInfo)
+            } else {
+                presenter.saveDeepLinkInfo(deepLinkInfo)
+                presenter.toOnBoarding()
             }
         }
     }
 
-    private fun showOnBoardingFragment() {
-        addFragment(
-            ScreenViewEvent.OnBoarding.screenName,
-            R.id.fragment_container,
-            allowStateLoss = true
-        ) {
-            chat.rocket.android.authentication.onboarding.ui.newInstance()
-        }
-    }
-
-    private fun showServerFragment(deepLinkInfo: LoginDeepLinkInfo) {
-        addFragment(
-            ScreenViewEvent.Server.screenName,
-            R.id.fragment_container,
-            allowStateLoss = true
-        ) {
-            chat.rocket.android.authentication.server.ui.newInstance()
+    private fun routeNoLink() {
```

Comment:
What's route no link means? 🤔

Reply:
This is the default for when there is no dynamic link associated with the new intent.

Reply:
@ear-dev Can you add a comment to make it clear to anyone reading that code or change the function name to something more meaningful? It will help us a lot here.